src/rpc/server.zig

```diff
@@ -47,485 +37,265 @@ pub const Server = struct {
-
-        server.wait_group.start();
-        errdefer server.wait_group.finish();
-
-        const new_hct = try HandleConnectionTask.createAndReceiveHead(server, conn);
-        errdefer new_hct.destroyAndClose();
-
-        server.thread_pool.schedule(ThreadPool.Batch.from(&new_hct.task));
-    }
 };
 
-const HandleConnectionTask = struct {
-    task: ThreadPool.Task,
-    server: *Server,
-    http_server: std.http.Server,
-    request: std.http.Server.Request,
-
-    fn createAndReceiveHead(
-        server: *Server,
-        conn: std.net.Server.Connection,
-    ) (std.http.Server.ReceiveHeadError || std.mem.Allocator.Error)!*HandleConnectionTask {
-        const allocator = server.allocator;
-
-        const hct_buf_align = @alignOf(HandleConnectionTask);
-        const hct_buf_size = initBufferSize(server.read_buffer_size);
-
-        const hct_buffer = try allocator.alignedAlloc(u8, hct_buf_align, hct_buf_size);
-        errdefer server.allocator.free(hct_buffer);
-
-        const hct: *HandleConnectionTask = std.mem.bytesAsValue(
-            HandleConnectionTask,
-            hct_buffer[0..@sizeOf(HandleConnectionTask)],
-        );
-        hct.* = .{
-            .task = .{ .callback = callback },
-            .server = server,
-            .http_server = std.http.Server.init(conn, getReadBuffer(server.read_buffer_size, hct)),
-            .request = try hct.http_server.receiveHead(),
-        };
-
-        return hct;
-    }
-
-    /// Does not release the connection.
-    fn destroyAndClose(hct: *HandleConnectionTask) void {
-        const allocator = hct.server.allocator;
-
-        const full_buffer = getFullBuffer(hct.server.read_buffer_size, hct);
-        defer allocator.free(full_buffer);
-
-        const connection = hct.http_server.connection;
-        defer connection.stream.close();
-    }
-
-    fn initBufferSize(read_buffer_size: usize) usize {
-        return @sizeOf(HandleConnectionTask) + read_buffer_size;
-    }
-
-    fn getFullBuffer(
-        read_buffer_size: usize,
-        hct: *HandleConnectionTask,
-    ) []align(@alignOf(HandleConnectionTask)) u8 {
-        const ptr: [*]align(@alignOf(HandleConnectionTask)) u8 = @ptrCast(hct);
-        return ptr[0..initBufferSize(read_buffer_size)];
-    }
-
-    fn getReadBuffer(
-        read_buffer_size: usize,
-        hct: *HandleConnectionTask,
-    ) []u8 {
-        return getFullBuffer(read_buffer_size, hct)[@sizeOf(HandleConnectionTask)..];
-    }
-
-    fn callback(task: *ThreadPool.Task) void {
-        const hct: *HandleConnectionTask = @fieldParentPtr("task", task);
-        defer hct.destroyAndClose();
+pub const WorkPool = union(enum) {
+    basic,
+    linux_io_uring: switch (LinuxIoUring.can_use) {
+        .yes, .check => *LinuxIoUring,
+        .no => noreturn,
+    },
 
-        const server = hct.server;
-        const logger = server.logger;
+    pub const LinuxIoUring = @import("server/linux_io_uring.zig").LinuxIoUring;
```

Comment:
can we stick to normal import rules (all at top)

Reply:
My contention there is that `LinuxIoUring` should be directly associated with `WorkPool`, as it should only ever be instantiated in relation to it; `WorkPool.LinuxIoUring` makes this clear, and makes it so one only needs to import `WorkPool` in order to gain access to it, rather than having to import both separately from the server namespace.

Reply:
makes sense i was meaning that we have `pub const LinuxIoUring = LinuxIoUring` and removing the @import